scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -0,0 +1,50 @@
+package com.twitter.scalding.beam_backend
+
+import com.stripe.dagon.Rule
+import com.twitter.scalding.Execution.{ToWrite, Writer}
+import com.twitter.scalding.typed._
+import com.twitter.scalding.{CFuture, Config, Execution, ExecutionCounters}
+import org.apache.beam.sdk.Pipeline
+import scala.concurrent.{ExecutionContext, Future}
+
+class BeamWriter(val beamMode: BeamMode) extends Writer {
+  override def start(): Unit = ()
+  override def finished(): Unit = ()
+
+  def getForced[T](conf: Config, initial: TypedPipe[T])(implicit cec: ExecutionContext): Future[TypedPipe[T]] = ???
+
+  def getIterable[T](conf: Config, initial: TypedPipe[T])(implicit cec: ExecutionContext): Future[Iterable[T]] = ???
+
+  override def execute(
+    conf: Config,
+    writes: List[ToWrite[_]])(implicit cec: ExecutionContext): CFuture[(Long, ExecutionCounters)] = {
+    import Execution.ToWrite._
+    val planner = BeamPlanner.plan(conf, beamMode.sources)
+    val phases: Seq[Rule[TypedPipe]] = OptimizationRules.standardMapReduceRules
+    val optimizedWrites = ToWrite.optimizeWriteBatch(writes, phases)
+    val pipeline = Pipeline.create(beamMode.pipelineOptions)
+
+    def rec(optimizedWrites: List[OptimizedWrite[TypedPipe, _]]): Unit = {
+      optimizedWrites match {
+        case Nil => Unit
+        case x :: xs =>
+          x match {
+            case OptimizedWrite(pipe, ToWrite.SimpleWrite(opt, sink)) => {
+              val pcoll = planner(opt).run(pipeline)
+              beamMode.sink(sink) match {
+                case Some(ssink) =>
+                  ssink.write(pipeline, conf, pcoll)
+                case _ => throw new Exception(s"unknown sink: $sink when writing $pipe")
+              }
+              rec(xs)
+            }
+            case _ => rec(xs)
+          }
+      }
+    }
+    rec(optimizedWrites)
+    CFuture.uncancellable(Future{
+      pipeline.run().waitUntilFinish()
+    }.map(_ => (0L, ExecutionCounters.empty)))
```

Comment:
are there no counters in Beam?

you don't want to return `0L` for each item. The contract is to return a new unique Long for each call to this function. You can keep an atomicLong and increment on each call, or something like that.